### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by markelog**

### CODE REVIEW

File: src/Traverse.js [hunk 1]
```diff
@@ -82,6 +86,21 @@ class ElementIndexByType {
         }
     }
 
+    values(value: string): Array<any> {
+        let result = [];
+        console.log(this._index);
```
Comment (hzoo) [inline src/Traverse.js]: Should this be removed?

**Comment by hzoo** [comment]
looks good?

File: src/Traverse.js [hunk 2]
```diff
@@ -21,6 +21,10 @@ export default class ElementSearchIndex {
         return this._tokenIndex.select(type);
     }
 
+    selectTokensByValue(value: string): Array<Token> {
```
Comment (mdevils) [inline src/Traverse.js]: This method already fails in JSCS for some tokens and will also fail for CST now. Selecting tokens by value without specifying type should not be possible. For instance, value of comment-token `/*+*/` is `+`, value of `//+` is also `+`, and TemplateStrings consist of several tokens with arbitary values.

File: src/Traverse.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 
-import type Element from '../../Element';
-import type Node from '../../Node';
-import type Token from '../../Token';
+import type Element from './elements/Element';
+import type Node from './elements/Node';
+import type Token from './elements/Token';
 
-export default class ElementSearchIndex {
+export default class Traversal {
```
Comment (mdevils) [inline src/Traverse.js]: Why would you rename it to `Traversal`? I think `ElementSearchIndex` means exactly what it does.

File: src/elements/types/Program.js [hunk 4]
```diff
@@ -5,18 +5,18 @@ import type Node from '../Node';
 import type Element from '../Element';
 import type ElementAssert from '../ElementAssert';
 import Statement from '../Statement';
-import ElementSearchIndex from './utils/ElementSearchIndex';
+import Traverse from '../../Traverse';
```
Comment (mdevils) [inline src/elements/types/Program.js]: `Traverse` but class is named `Traversal`

File: src/Traverse.js (hunk 1 above)
Comment (mdevils) [inline src/Traverse.js]: Yes, this `console.log` should be removed.

**Comment by mdevils** [comment]
Thank you @markelog. I've added a pair of comments.

File: src/Traverse.js (hunk 2 above)
Comment (markelog) [inline src/Traverse.js]: mm, well you have the same behaviour for other methods as well. Those methods could be useful, it is pretty much analogues to CSS selector `:contains(...)`. You would rarely use it, but you might be glad it is there.

File: src/elements/types/Program.js (hunk 4 above)
Comment (markelog) [inline src/elements/types/Program.js]: Fixed

File: src/Traverse.js (hunk 3 above)
Comment (markelog) [inline src/Traverse.js]: In dom analogues api's called in same manner, http://www.w3.org/TR/ElementTraversal/#interface-elementTraversal, https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll
https://api.jquery.com/category/traversing/

Comment (markelog) [inline src/Traverse.js]: Although i could change it back if you insist :)

**Pull request merged**